### PR TITLE
Fix the returning null value when only submitterParameter and proceeding from console output page

### PR DESCRIPTION
**Pull request by tangkun75**
Like the below step:
...
def x = input message: 'Approve or not', submitter: 'admin,dev', submitterParameter: 'my-approval'
echo x
...
When you click "Proceed" from the console output page, "x will be null in fact, but we expect x should be admin or dev.

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStepExecution.java [hunk 1]
```diff
@@ -71,7 +71,7 @@ public boolean start() throws Exception {
         node.addAction(new PauseAction("Input"));
 
         String baseUrl = '/' + run.getUrl() + getPauseAction().getUrlName() + '/';
-        if (input.getParameters().isEmpty()) {
+        if (input.getParameters().isEmpty() && (input.getSubmitterParameter() == null || input.getSubmitterParameter().isEmpty())) {
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStepExecution.java]: Empty string trimming should be done in `setSubmitterParameter` if it is not already.

Comment (tangkun75) [inline src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStepExecution.java]: Thanks for let me know the 
https://issues.jenkins-ci.org/browse/JENKINS-40594, @jglick 
Yes, this change could fix the JENKINS-40594.

Comment (tangkun75) [inline src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStepExecution.java]: I try to understand your point about setSubmitterParameter method. Could you supply more detail?
Thanks,

**Comment by tangkun75** [comment]
@jglick ,
I've submitted the changes for JENKINS-40594 and with @Issue annotation. Please review it.

File: src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStepExecution.java [hunk 2]
```diff
@@ -71,7 +71,8 @@ public boolean start() throws Exception {
         node.addAction(new PauseAction("Input"));
 
         String baseUrl = '/' + run.getUrl() + getPauseAction().getUrlName() + '/';
-        if (input.getParameters().isEmpty()) {
+        //JENKINS-40594 submitterParameter does not work without at least one actual parameter
+        if (input.getParameters().isEmpty() && (input.getSubmitterParameter() == null || input.getSubmitterParameter().isEmpty())) {
```
Comment (jglick) [inline src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStepExecution.java]: Due to [this trimming](https://github.com/jenkinsci/pipeline-input-step-plugin/blob/ab59e21435764953138782a0abdb2e13f989d28c/src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStep.java#L84) it should suffice to say

```java
if (input.getParameters().isEmpty() && input.getSubmitterParameter() == null) {
```

**Comment by jglick** [comment]
> When you click "Proceed" from the console output page, "x will be null in fact, but we expect x should be admin or dev.

Does that not contradict the existing test, which seems to verify that JENKINS-40594 does not exist? I am confused by the status of this bug.

File: src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStepExecution.java (hunk 2 above)
Comment (tangkun75) [inline src/main/java/org/jenkinsci/plugins/workflow/support/steps/input/InputStepExecution.java]: Thanks for your help, I did tests again with your changes with below 3 test cases. It works well.
Yes, your guess is correct: this issue is caused by the console hyperlink in UI.
===========================================================================
def x = input message: 'Approve or not: case 1', submitter: 'admin,dev', submitterParameter: 'my-approval'
echo x
x = input message: 'Approve or not: case 2', submitterParameter: 'my-approval'
echo x
x = input message: 'Approve or not: case 3', submitter: '', submitterParameter: 'my-approval'
echo x
================ Here is output of console ========================
Started by user admin
[Pipeline] input
Input requested
Approved by admin
[Pipeline] echo
admin
[Pipeline] input
Input requested
Approved by admin
[Pipeline] echo
admin
[Pipeline] input
Input requested
Approved by admin
[Pipeline] echo
admin
[Pipeline] End of Pipeline
Finished: SUCCESS

**Comment by tangkun75** [comment]
Yes, your guess is correct: this issue is caused by the console hyperlink in UI.

**Pull request merged**